src/nvim/screen.c

```diff
@@ -5332,11 +5333,29 @@ static void win_redr_custom(win_T *wp, bool draw_winbar, bool draw_ruler)
     attr = (wp == curwin) ? HL_ATTR(HLF_WBR) : HL_ATTR(HLF_WBRNC);
     maxwidth = wp->w_width_inner;
     use_sandbox = was_set_insecurely(wp, "winbar", 0);
+
+    if (wp->w_winbar_height && wp->w_winbar_click_defs_size < (size_t)maxwidth) {
```

Comment:
a terse comment can help a lot when reading these big walls of code.
```suggestion
    // prepare clickable areas (?)
    if (wp->w_winbar_height && wp->w_winbar_click_defs_size < (size_t)maxwidth) {
```